pod.go

```diff
@@ -525,12 +525,48 @@ func (p *Pod) createSetStates() error {
 	return nil
 }
 
+func selectAssets(podConfig *PodConfig) error {
+	kernel, err := newAsset(podConfig, kernelAsset)
+	if err != nil {
+		return err
+	}
+
+	image, err := newAsset(podConfig, imageAsset)
+	if err != nil {
+		return err
+	}
+
+	if kernel != nil && kernel.path != "" {
```

Comment:
This test confused me the first time I read it. Might be worth a comment on the function to state that it modifies the podConfig if explicit assets have been specified, but otherwise uses the default values. This would be slightly clearer if the `if` tests had an `else` specifying something like:

```
else {
    virtLog.Debug("using default kernel asset as no custom value specified")
}
```

In fact, if you added a `func (a *asset) override(podConfig *PodConfig) override`, couldn't you compress these 2 blocks into:

```
for _, a := range []asset{image, kernel{
  // this could handle a==nil and also call asset.valid()
  asset.override(podconfig)
}
```